### PR TITLE
add model/controller for /api/articles/:article_id/comments and relev…

### PR DESCRIPTION
**Pull request by DeanDevine**
…ant tests, add error handling for 400, 404, update endpoints.json with path

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -134,3 +134,65 @@ describe('GET /api/articles', () => {
...
+describe('GET /api/articles/:article_id/comments', () => {
+    test('status:200, responds with a JSON object containing an array of comment objects for the given article_id of which each object has the following properties: comment_id, votes, created_at, author, body, article_id', () => {
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+
+            const { comments } = body;
+
+            expect(comments).toHaveLength(11);
```
Comment (johnhutcheon) [inline __tests__/app.test.js:147]: Nice work checking the length before using a forEach 👍

File: __tests__/app.test.js [hunk 2]
```diff
@@ -134,3 +134,65 @@ describe('GET /api/articles', () => {
...
+        return request(app)
+        .get('/api/articles/1/comments')
+        .expect(200)
+        .then(({body}) => {
+
+            const { comments } = body;
+
+            expect(comments).toHaveLength(11)
+
+            expect(comments).toBeSorted('created_at', { ascending: true })
```
Comment (johnhutcheon) [inline __tests__/app.test.js:174]: Thanks for implementing jest sorted as requested! 🙌

File: __tests__/app.test.js [hunk 3]
```diff
@@ -134,3 +134,65 @@ describe('GET /api/articles', () => {
...
+
+        })
```
Comment (johnhutcheon) [inline __tests__/app.test.js:176]: Could we add one more test for a 200 please? What would we expect to happen if it’s a valid article ID but it has no comments?

File: models/model-articles.js [hunk 4]
```diff
@@ -32,3 +32,25 @@ exports.selectArticle = (article_id) => {
...
+    created_at, 
+    author, 
+    body, 
+    article_id 
+    FROM comments 
+    WHERE article_id = $1
+    ORDER BY created_at ASC`, [article_id])
+    .then(({rows}) => {
+        if (!rows.length) {
+            return Promise.reject({ status: 404, msg: "Not Found" })
```
Comment (johnhutcheon) [inline models/model-articles.js]: Nicely done!

**Review by johnhutcheon — CHANGES_REQUESTED** [review]
Hi Dean, really good work! Could you just add the extra test as per my comments please? Thanks!

File: __tests__/app.test.js [hunk 5]
```diff
@@ -175,23 +175,25 @@ describe('GET /api/articles/:article_id/comments', () => {
...
-        .get('/api/articles/9999/comments')
-        .expect(404)
+        .get('/api/articles/:not_article_id/comments')
+        .expect(400)
         .then(({body}) => {
 
-            expect(body.msg).toBe('Not Found')
+            expect(body.msg).toBe('Invalid Input')
 
         })
```
Comment (MKD01) [inline __tests__/app.test.js:199]: Here you have removed the test for when a valid id is given but does not exist, this was a really good test and should be in this ticket.

**Review by MKD01 — CHANGES_REQUESTED** [review]
Great work so far, you are almost there with this ticket but still missing a test, once you have added this back in please re-submit this as a new pr 👍

File: __tests__/app.test.js [hunk 6]
```diff
@@ -134,3 +134,90 @@ describe('GET /api/articles', () => {
...
+            const { comments } = body;
+
+            expect(comments).toHaveLength(11)
+
+            expect(comments).toBeSorted('created_at', { ascending: true })
+
+        })
+    })
+    test('status:200, returns an array containing an article object which contains no comments when article_id is valid but does not have any comments', () => {
+        return request(app)
```
Comment (daviidbartlett) [inline __tests__/app.test.js:179]: this is off behaviour to design. I'm request all the comments associated with an article.
Assuming the article exists, I'm expect an array of all of the comments. If the there were no comments, that array would be empty. (Still a 200 though!)

**Review by roisinoreilly — APPROVED** [review]
nicely refactored :) PR approved merge when ready

**Pull request merged**